compress/compress.go

```diff
@@ -29,3 +32,12 @@ func Compress(r io.Reader) (io.Reader, error) {
 	}
 	return &buf, nil
 }
+
+// IsCompressed detects whether the input stream is compressed.
+func IsCompressed(r io.Reader) (bool, error) {
```

Comment:
This consumes the first two bytes of the reader, so downstream readers can't get it. Is that okay?

ISTR some Go stdlib thing does something like this, but returns a new `io.Reader` with the buffered bytes prepended.

So instead of `c, err := IsCompressed(r)` you'd have to write `r, c, err = IsCompressed(r)` which is a pain in and of itself.

Reply:
Yeah, at first I had a version of this that returned a reader so we could continue to read the input after deciding if it was compressed or not.

But then I realized the way this code is structured now, we don't need to read the layer anymore after (until much later, in a different function).

Maybe I should make it take a ReadCloser, and close it in this function to be explicit about that?

Reply:
Yeah in that case being explicit about what modifications you make to the reader seems really important. I'm curious how that later far-away read works if the first two bytes of the data have been consumed?

Reply:
I made it return an un-advanced reader.

In this case we're just throwing away the reader after determining if it's compressed, so we ignore the newly returned one.

The compression would have been performed in another function, after opening the layer again.